io/net/bonding.py

```diff
@@ -95,7 +95,10 @@ def setUp(self):
             for ipaddr, interface in zip(self.ipaddr, self.host_interfaces):
                 networkinterface = NetworkInterface(interface, self.localhost)
                 try:
-                    networkinterface.nm_flush_ipaddr()
+                    if self.detected_distro.name == "SuSE":
```

Comment:
I think the latest SUSE version uses nmcli ..can you confirm and handle for new and only rhel and sles distro

Reply:
@abdhaleegit Yes, SUSE does support nmcli/NetworkManager but not as default package, but we can install it by configuring repo's / tar ball separately, as "wicked" or "network" services are act as default services during the OS bringup. 
but in RHEL "NetworkManager" is default service for Network management operations.

1. We can automate to use only "nmcli" by installing from source tar, but the caveat what we may face is , since wicked or network service maintains network operations by-default, with Network manager installation we will have to disable wicked and switch to Network manager, which causes network loss. [ some times it may required reboot also]

2. The main reason why we implemented "nm_flush_address" , because in RHEL when we use "ip addr flush" for statically configured IP addresses , addresses are not getting deleted. but same logic works for SLES. [ it will delete static and on-fly addresses of interface]

so since most of the clean up problem were fenced around "ip addr flush" which has problem with RH, we are implemented nmcli based flushing the IP address.